src/libopensc/card-piv.c

```diff
@@ -4686,6 +4754,20 @@ piv_decipher(sc_card_t *card, const u8 * data, size_t datalen, u8 * out, size_t
 {
 	SC_FUNC_CALLED(card->ctx, SC_LOG_DEBUG_VERBOSE);
 
+	piv_private_data_t * priv = PIV_DATA(card);
+	u8 touch_policy;
+	const char *title = "Touch your Yubikey to continue";
+
+	piv_yk_get_metadata(card, priv->key_ref, NULL, &touch_policy);
+	switch (touch_policy) {
+		case 0x02:
+			sc_notify(title, "Touching the token is required for deciphering");
+			break;
+		case 0x03:
+			sc_notify(title, "Touching the token is required for deciphering (needed again after 15 seconds)");
+			break;
+	}
```

Comment:
Now, this code is the same for both sign/decipher. Would it be possible to somehow share it to avoid copy&paste?

Reply:
Yes, if we accept that the notification message is the same for both cases

Reply:
Oh, i missed the difference in the message. But given that its just one word, I think it could be an argument.

Reply:
i reworded and unified the message